### PR TITLE
Refactor Jacobian methods and operators

### PR DESCRIPTION
**Pull request by domire8**
## Description

- #99 

Wow, I'm so pleased by how that refactor turned out. Many deleted lines, no big breaking changes in robot model or controllers, overall a big success and improvement in my opinion! Everything according to the parent issue.

Maybe you can have a look at the docstrings, if they make sense.

## Review guidelines

Estimated Time of Review: 20 minutes
Review by commit

### CODE REVIEW

**Comment by domire8** [comment]
I will fix the Python bindings in a follow up PR, there might be more to be done there

File: source/state_representation/include/state_representation/space/Jacobian.hpp [hunk 1]
```diff
@@ -252,19 +275,12 @@ class Jacobian : public State {
...
   /**
-   * @brief Overload the * operator with another Jacobian
-   * @param jacobian The Jacobian to multiply with
-   * @return The current Jacobian multiplied by the one in parameter
-   */
-  Eigen::MatrixXd operator*(const Jacobian& jacobian) const;
-
-  /**
-   * @brief Overload the * operator with an arbitrary matrix on the left side
+   * @brief Overload the * operator with an 6x6 matrix on the left side
```
Comment (eeberhard) [inline source/state_representation/include/state_representation/space/Jacobian.hpp:278]:
```suggestion
   * @brief Overload the * operator with a 6x6 matrix on the left side
```

File: source/state_representation/include/state_representation/space/Jacobian.hpp [hunk 2]
```diff
@@ -222,27 +236,36 @@ class Jacobian : public State {
...
+   * @return The result of the matrix multiplication
    */
-  Eigen::MatrixXd solve(const Eigen::MatrixXd& matrix) const;
+  Eigen::MatrixXd pseudoinverse(const Eigen::MatrixXd& matrix) const;
 
   /**
-   * @brief Solve the system dX = J*dq to obtain dq which is more efficient than multiplying with the pseudo-inverse
-   * @param dX The Cartesian twist to multiply with
+   * @brief Transform the given Cartesian twist to joint space by multiplication with the pseudoinverse of the
+   * Jacobian matrix
```
Comment (eeberhard) [inline source/state_representation/include/state_representation/space/Jacobian.hpp:250]: nitpick: I don't like "the given" in docstrings. It's over-specifying something generic. Also, the rest of the sentence can be made more compact and accurate as follows:
```suggestion
   * @brief Transform a Cartesian twist to joint space by pre-multiplying the Jacobian pseudoinverse
```
Same suggestion to apply to CartesianWrench transform.

**Review by eeberhard — COMMENTED** [review]
This looks great to me! I commented on docstrings as suggested, but overall I'm happy with the refactor direction

**Comment by domire8** [comment]
Thanks, I appreciate it

**Pull request merged**